nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -29,6 +29,22 @@ let
     inherit (efi) efiSysMountPoint canTouchEfiVariables;
 
     memtest86 = if cfg.memtest86.enable then pkgs.memtest86-efi else "";
+
+    netbootxyz = if cfg.netbootxyz.enable then pkgs.netbootxyz-efi else "";
+
+    copyExtraFiles = pkgs.writeShellScript "copy-extra-files" ''
+      empty_file=$(mktemp)
```

Comment:
nitpick if you're up to it: we've since learned about the `pkgs.emptyFile` trivial builder. might be better to use that, and solve the TMPDIR problem with the installer?

apart from that, LGTM

Reply:
https://github.com/NixOS/nixpkgs/blob/1783cfde53a58b1fa38449e47f45dd65091de7ae/nixos/modules/installer/tools/nixos-enter.sh#L103-L104

I think the `TMPDIR` needs to be fixed irrespective of whether this PR uses it or not, as the `nixos-enter` script creates `/tmp` but then won't even use it if `TMPDIR` is set.

My current preference is to keep `mktemp` to dogfood the `nixos-enter` command :)

Reply:
how did you find that? just tested an installer run (on 21.11 though, not unstable) and found that TMPDIR wasn't set.

Reply:
interactions between different sides of the same chroot sure can be confusing. :+1: